### DISCUSSION TITLE
Port bit7z to vcpkg

### DISCUSSION THREAD

**Discussion by msclock** [rikyoz/bit7z#243] [Show and tell]
Hi, I just port bit7z to vcpkg in https://github.com/microsoft/vcpkg/pull/40889. 

```bash
$ vcpkg search bit7z
bit7z                    4.0.8            A C++ static library offering a clean and simple interface to the 7-zip sh...
bit7z[regex-matching]                     Enables the support for extracting files matching regular expressions
The result may be outdated. Run `git pull` to get the latest results.
If your port is not listed, please open an issue at and/or consider making a pull request.  -  https://github.com/Microsoft/vcpkg/issues
```

If there was anything wrong, please comment!

**Comment by rikyoz** [comment]
Hi!
Thank you for the port!

I've noticed some patches in your PR. 
I'm not familiar with vcpkg, but I guess that the changes they introduce are required to make the library work via vcpkg, right?

Also, there are some other build options that might be useful to add.
In particular, [`BIT7Z_USE_LEGACY_IUNKNOWN`](https://github.com/rikyoz/bit7z/wiki/Building-the-library#unix-only-options) (Unix-only), [`BIT7Z_LINK_LIBCPP`](https://github.com/rikyoz/bit7z/wiki/Building-the-library#clang-only-options) (Clang-only), and  [`BIT7Z_STATIC_RUNTIME`](https://github.com/rikyoz/bit7z/wiki/Building-the-library#msvc-only-options) (MSVC-only).

Other than that, I don't see any specific problem.
Thank you again! 🙏

**Reply by msclock** [reply]
> I've noticed some patches in your PR.
> I'm not familiar with vcpkg, but I guess that the changes they introduce are required to make the library work via vcpkg, right?

Yep, those patches make it to compile and install correctly for vcpkg integration.

> In particular, [BIT7Z_USE_LEGACY_IUNKNOWN](https://github.com/rikyoz/bit7z/wiki/Building-the-library#unix-only-options) (Unix-only), [BIT7Z_LINK_LIBCPP](https://github.com/rikyoz/bit7z/wiki/Building-the-library#clang-only-options) (Clang-only), and [BIT7Z_STATIC_RUNTIME](https://github.com/rikyoz/bit7z/wiki/Building-the-library#msvc-only-options) (MSVC-only).

I would add these options right now. And I notice port 7zip in vcpkg didn't use Z7_USE_VIRTUAL_DESTRUCTOR_IN_IUNKNOWN for 7-zip 23.01+. So I would not like to provide BIT7Z_USE_LEGACY_IUNKNOWN as a feature so far.

**Reply by msclock** [reply]
After that pr merged, we could inform customers the new usage. 😀

**Reply by rikyoz** [reply]
> I would add these options right now. And I notice port 7zip in vcpkg didn't use Z7_USE_VIRTUAL_DESTRUCTOR_IN_IUNKNOWN for 7-zip 23.01+. So I would not like to provide BIT7Z_USE_LEGACY_IUNKNOWN as a feature so far.

Uhm ok. I guess that Unix users of the port will have to use 7-Zip 23.01+ libraries then.

> After that pr merged, we could inform customers the new usage. 😀

Sure, I'll add instructions on how to use the library via vcpkg in the README and Wiki.

**Reply by msclock** [reply]
bit7z port has been merged by the downstream vcpkg. As we discused in the downstream there is only one feature `regex-matching` in the features list and I have updated the discussion to conform to it.
